src/ucp/core/ucp_mm.c

```diff
@@ -119,22 +118,18 @@ ucs_status_t ucp_mem_rereg_mds(ucp_context_h context, ucp_md_map_t reg_md_map,
                 continue;
             }
 
-            base_address = address;
-            reg_length   = length;
-
-            if (context->config.ext.reg_whole_alloc_bitmap & UCS_BIT(mem_type)) {
-                ucp_memory_detect_internal(context, address, length, &mem_info);
-                base_address = mem_info.base_address;
-                reg_length   = mem_info.alloc_length;
-            }
+            reg_address = address;
+            reg_length  = length;
+            ucp_context_get_reg_memory(context, &reg_address, &reg_length,
+                                       mem_type);
```

Comment:
this func relies on pre-initialized values of `address` and `length`, which seems to be error-prone.
Maybe better pass initial `address` and `length` as parameters to `ucp_context_get_reg_memory`?
Also this would reduce duplication (initializing `address` and `length`)

Reply:
fixed